### PR TITLE
Feature/visualization improvements

### PR DESCRIPTION
**Pull request by Jammy2211**
Makes a number of improvements to how visualization is performed to make better figures, and cleans up unsed parts of the source code.

### CODE REVIEW

File: autoarray/geometry/geometry_util.py [hunk 1]
```diff
@@ -740,3 +740,39 @@ def grid_pixel_centres_2d_from(
             )
 
     return grid_pixels_2d
+
+
+def extent_symmetric_from(extent : Tuple[float, float, float, float]) -> Tuple[float, float, float, float]:
```
Comment (rhayes777) [inline autoarray/geometry/geometry_util.py:745]: Could be worth defining an extent type

```python
Extent = Tuple[float, float, float, float]

File: autoarray/plot/mat_plot/one_d.py [hunk 2]
```diff
@@ -201,40 +205,69 @@ def plot_yx(
...
+            text_manual_list = []
+
+            for key, value in text_manual_dict.items():
+
+                text_manual_list.append(
+                    Text(x=0.65, y=y, s=f"{key} : {value}", c="b", transform=ax.transAxes)
+                )
+                y = y - 0.05
+
+            [text.set() for text in text_manual_list]
```
Comment (rhayes777) [inline autoarray/plot/mat_plot/one_d.py:269]: Why make a list? Just call text.set() in the loop?

File: autoarray/plot/mat_plot/two_d.py [hunk 3]
```diff
@@ -435,8 +432,9 @@ def plot_mapper(
         mapper: Union[MapperRectangularNoInterp, MapperVoronoiNoInterp],
         visuals_2d: Visuals2D,
         auto_labels: AutoLabels,
-        interpolate_to_uniform: bool = True,
-        source_pixelization_values=None,
+        interpolate_to_uniform: bool = False,
+        source_pixelization_values : np.ndarray = None,
```
Comment (rhayes777) [inline autoarray/plot/mat_plot/two_d.py:436]: Optional

File: autoarray/plot/mat_plot/two_d.py [hunk 4]
```diff
@@ -465,15 +464,16 @@ def plot_mapper(
...
             )
 
     def _plot_rectangular_mapper(
         self,
         mapper: MapperRectangularNoInterp,
         visuals_2d: Visuals2D,
         auto_labels: AutoLabels,
-        interpolate_to_uniform: bool = True,
-        source_pixelization_values=None,
+        source_pixelization_values : np.ndarray = None,
```
Comment (rhayes777) [inline autoarray/plot/mat_plot/two_d.py:475]: Optional

File: autoarray/plot/wrap/base/colorbar.py [hunk 5]
```diff
@@ -33,37 +37,106 @@ def __init__(
...
-                "both the values and labels, or neither."
+            cb = plt.colorbar(ticks=manual_tick_values, ax=ax, **self.config_dict)
+            cb.ax.set_yticklabels(
+                labels=manual_tick_labels, va=self.manual_alignment or "center"
             )
 
+        # if manual_tick_labels is None:
+        #     manual_tick_labels = ["{:.1f}".format(ytick) for ytick in cb.ax.get_yticks()]
+        #
+        # cb.ax.set_yticklabels(
```
Comment (rhayes777) [inline autoarray/plot/wrap/base/colorbar.py:131]: Commented code

File: autoarray/plot/wrap/base/colorbar.py [hunk 6]
```diff
@@ -82,19 +155,33 @@ def set_with_color_values(self, cmap: str, color_values: np.ndarray, ax=None):
...
                 **self.config_dict,
             )
             cb.ax.set_yticklabels(
-                labels=self.manual_tick_labels, va=self.manual_alignment or "center"
+                labels=manual_tick_labels, va=self.manual_alignment or "center"
             )
 
+        # if manual_tick_labels is None:
+        #     manual_tick_labels = ["{:.1f}".format(ytick) for ytick in cb.ax.get_yticks()]
+        #
```
Comment (rhayes777) [inline autoarray/plot/wrap/base/colorbar.py:182]: Commented code

File: autoarray/plot/wrap/base/ticks.py [hunk 7]
```diff
@@ -32,16 +32,17 @@ def __init__(
...
         super().__init__(**kwargs)
 
+        self.manual_extent_factor = manual_extent_factor
         self.manual_values = manual_values
         self.manual_units = manual_units
-        self.suffix = suffix
+        self.manual_suffix = manual_suffix
 
-    def tick_values_from(self, min_value: float, max_value: float) -> np.ndarray:
+    def tick_values_from(self, min_value: float, max_value: float, is_for_1d_plot : bool = False) -> np.ndarray:
```
Comment (rhayes777) [inline autoarray/plot/wrap/base/ticks.py:45]: Could just explicitly pass the suffix?

File: autoarray/plot/wrap/base/ticks.py [hunk 8]
```diff
@@ -89,36 +109,41 @@ def tick_values_in_units_from(
...
+            return self.manual_suffix
+
+        if yunit is not None:
+            return yunit
+
+        if units.ticks_label is not None:
+            return units.ticks_label
+
+        if units is None:
+            return conf.instance["visualize"]["general"]["units"]["unscaled_symbol"]
```
Comment (rhayes777) [inline autoarray/plot/wrap/base/ticks.py:139]: Assign units config to a variable

```python
units_conf = conf.instance["visualize"]["general"]["units"]

File: autoarray/plot/wrap/base/ticks.py [hunk 9]
```diff
@@ -156,11 +183,34 @@ def set(
...
+            ticks = np.logspace(np.log10(min_value), np.log10(max_value), number)
+
+            plt.ylim(min_value, max_value)
+
+            labels = self.tick_values_in_units_from(
+                tick_values=ticks,
+                units=units,
+                round_value=False
+            )
+            labels = ['{:.0e}'.format(label) for label in labels]
```
Comment (rhayes777) [inline autoarray/plot/wrap/base/ticks.py:202]: You can do advanced formatting with fstrings

**Review by rhayes777 — APPROVED** [review]

File: test_autoarray/config/visualize.yaml [hunk 10]
```diff
@@ -13,8 +13,20 @@ general:
...
+    9: (3, 3)                         # The shape of subplots for a figure with 9 (or less than the above value) of subplots.
+    12: (3, 4)                        # The shape of subplots for a figure with 12 (or less than the above value) of subplots.
+    16: (4, 4)                        # The shape of subplots for a figure with 16 (or less than the above value) of subplots.
+    20: (4, 5)                        # The shape of subplots for a figure with 20 (or less than the above value) of subplots.
+    36: (6, 6)                        # The shape of subplots for a figure with 36 (or less than the above value) of subplots.
+    49: (7, 7)                        # The shape of subplots for a figure with 49 (or less than the above value) of subplots.
+    64: (8, 8)                        # The shape of subplots for a figure with 64 (or less than the above value) of subplots.
+    81: (9, 9)                        # The shape of subplots for a figure with 81 (or less than the above value) of subplots.
+    100: (10, 10)                     # The shape of subplots for a figure with 100 (or less than the above value) of subplots.
```
[the remark this answers is not in the document]
Comment (Jammy2211) [reply test_autoarray/config/visualize.yaml:29]: This is to enable customization across different projects.

**Pull request merged**